### ISSUE TITLE
Training issues in Syncnet

### ISSUE THREAD

**Issue by QUTGXX** [Rudrabha/Wav2Lip#16]
Hi, thanks for sharing. When I tried to train the dataset (made by myself followed by LRS2 dataset) with 'wav2lip_train.py', it failed. And it told me that 'Expected more than 1 value per channel when training, got input size torch.Size([1, 512, 1, 1])' . Could you please help me to address this issue? The following image shows this error.

![屏幕截图](https://user-images.githubusercontent.com/36947214/92373810-76614880-f131-11ea-8d18-ce4508be9549.png)

**Comment by prajwalkr** [comment]
Hello, thanks for trying out the code. We had not encountered this error. Can you print the shape of the `face_sequences` variable just before line 56 in forward function of syncnet.py?

**Comment by QUTGXX** [comment]
@prajwalkr 

> Hello, thanks for trying out the code. We had not encountered this error. Can you print the shape of the `face_sequences` variable just before line 56 in forward function of syncnet.py?

Thanks for your quick reply. The following images are the shape of face_sequences and audio features.

![屏幕截图_2](https://user-images.githubusercontent.com/36947214/92382523-5173d200-f13f-11ea-8c75-a25e458a12c8.png)
![屏幕截图_1](https://user-images.githubusercontent.com/36947214/92382540-5769b300-f13f-11ea-925e-69446cef8d2b.png)

**Comment by prajwalkr** [comment]
Can you please try once with `batch size > 1`?

**Comment by QUTGXX** [comment]
> Can you please try once with `batch size > 1`?

Actually,  the batch size of previous trying is 16.

**Comment by prajwalkr** [comment]
> Actually, the batch size of the previous trying is 16

The first dimension of the printed shape is equal to `1`. So, the batch size passed to the model is not 16. Ensure that the dataset is correctly preprocessed.

> the parameter of 'syncnet_wt' has changed from 0.0 to 0.05 in 'hparams.py'.

Did you change it? Our code does not change it to 0.05 by default. Also, please ensure you are using the latest version of the repo.

**Comment by QUTGXX** [comment]
> > Actually, the batch size of the previous trying is 16
> 
> The first dimension of the printed shape is equal to `1`. So, the batch size passed to the model is not 16. Ensure that the dataset is correctly preprocessed.
> 
> > the parameter of 'syncnet_wt' has changed from 0.0 to 0.05 in 'hparams.py'.
> 
> Did you change it? Our code does not change it to 0.05 by default. Also, please ensure you are using the latest version of the repo.

Yes, I changed it because the sync loss is 0.0 when I tried to train the model. If I did not change this parameter, the code works well. Can u help me to explain why the loss is zero and how I can try the model with my own dataset?

**Comment by prajwalkr** [comment]
For the initial stages of training, face reconstruction is important, so sync loss is set to zero to ensure faster convergence. After a point, the sync loss weight is automatically set to a non-zero value in the code itself. So, you do not have to change anything.

**Comment by QUTGXX** [comment]
> For the initial stages of training, face reconstruction is important, so sync loss is set to zero to ensure faster convergence. After a point, the sync loss weight is automatically set to a non-zero value in the code itself. So, you do not have to change anything.

Got it, thanks for your help. But can u please explain which point and then it would be set to a non-zero value?

**Comment by prajwalkr** [comment]
when the eval sync loss goes below a certain threshold: https://github.com/Rudrabha/Wav2Lip/blob/9e98706b60041149edb51b470a0aba8af73b845f/wav2lip_train.py#L253

**Comment by QUTGXX** [comment]
> when the eval sync loss goes below a certain threshold:
> 
> https://github.com/Rudrabha/Wav2Lip/blob/9e98706b60041149edb51b470a0aba8af73b845f/wav2lip_train.py#L253

Got it!

**Comment by NKdryer** [comment]
Hi!
How do you fix this ValueError?
@QUTGXX

**Comment by HaSaKiYasuooo** [comment]
> 你好！ 如何修复这个值错误？ @QUTGXX

我也遇到了这个问题，请问如和修复它

**Comment by 1jianyao** [comment]
I have been troubled by this problem for a long time until I added two videos to the training set.

train.txt
dataset_preprocessed/dataset_yao/1/
dataset_preprocessed/dataset_yao/3/
